src/components/DropMenu/DropMenu.jsx

```diff
@@ -0,0 +1,468 @@
+					if (focusedIndex < _.size(flattenedOptionsData) - 1) {
+						e.preventDefault();
+						onFocusNext(e);
+					}
+				} else {
+					e.preventDefault();
+					onFocusNext(e);
+				}
+			}
+		} else {
+			if (_.includes([KEYCODE.Space, KEYCODE.ArrowDown], e.keyCode)) {
+				e.preventDefault();
+				onExpand(e);
+			}
+		}
+	},
+
+	handleClick(e) {
+		const {
+			isExpanded,
+			onExpand,
+			onCollapse
+		} = this.props;
+
+		if (isExpanded) {
+			onCollapse(e);
+		} else {
+			onExpand(e);
+		}
+	},
+
+	handleChangeBounds(type) {
+		const {
+			onBelowFold,
+			onAboveFold
+		} = this.props;
+		if (type === ContextMenu.BELOW_FOLD) {
+			onBelowFold();
+		} else if (type === ContextMenu.ABOVE_FOLD) {
+			onAboveFold();
+		}
+	},
+
+	handleMouseFocusOption(optionIndex, optionProps) {
+		const {
+			focusedIndex,
+			onFocusOption
+		} = this.props;
+
+		this.setState({
+			isMouseTriggered: true
+		});
+
+		if (!optionProps.isDisabled && focusedIndex !== optionIndex) {
+			onFocusOption(optionIndex);
+		}
+	},
+
+	handleSelectOption(optionIndex, optionProps) {
+		const {
+			onSelect
+		} = this.props;
+
+		if (!optionProps.isDisabled) {
+			onSelect(optionIndex);
+		}
+	},
+
+	renderOption(optionProps, optionIndex, isGrouped) {
+		const {
+			selectedIndices,
+			focusedIndex,
+		} = this.props;
+
+		const {
+			isMouseTriggered
+		} = this.state;
+
+		const {
+			isDisabled
+		} = optionProps;
+
+		const isFocused = optionIndex === focusedIndex;
+		const isSelected = _.includes(selectedIndices, optionIndex);
+
+		return (
+			<div
+				className={boundClassNames(
+					'&-Option', {
+					'&-Option-is-grouped': isGrouped,
+					'&-Option-is-focused': isFocused,
+					'&-Option-is-selected': isSelected,
+					'&-Option-is-disabled': isDisabled,
+					'&-Option-is-null': _.isNull(optionIndex)
+				})}
+				onMouseMove={() => this.handleMouseFocusOption(optionIndex, optionProps)}
+				onClick={() => this.handleSelectOption(optionIndex, optionProps)}
+				ref={(optionDOMNode)=> {
+					if (isFocused && !isMouseTriggered) {
+						scrollParentTo(optionDOMNode);
+					}
+				}}
+				key={'DropMenuOption' + optionIndex}
+			>
+				{optionProps.children}
+			</div>
+		);
+	},
+
+	componentWillMount() {
+		// preprocess the options data before rendering
+		this.setState(this.getPreprocessedOptionData(this.props));
+	},
+
+	componentWillReceiveProps(nextProps) {
+		// only preprocess options data when it changes (via new props) - better performance than doing this each render
+		this.setState(this.getPreprocessedOptionData(nextProps));
+	},
+
+	render() {
+		const {
+			className,
+			style,
+			isDisabled,
+			isExpanded,
+			direction,
+			portalId,
+			onCollapse
+		} = this.props;
+
+		const {
+			optionGroups,
+			ungroupedOptionData,
+			optionGroupDataLookup,
+			nullOptions
+		} = this.state;
+
+		const controlProps = _.first(DropMenu.Control.findInAllAsProps(this.props));
+
+		return (
+			<div className={boundClassNames('&', '&-base', {
```

Comment:
Actually, after looking at `ContextMenu.FlyOut`, that's basically the root element.